pkg/gadgets/internal/tchandler/tc.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !withoutebpf
+
+package tchandler
+
+import (
+	"errors"
+	"fmt"
+	"net"
+
+	"github.com/cilium/ebpf"
+	"github.com/florianl/go-tc"
+	tccore "github.com/florianl/go-tc/core"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	flagDirectAction = uint32(0x1)
+	filterInfo       = uint32(0x00010300) // priority << 16 | proto
```

Comment:
Could you add some explanation about this? Is the priority & proto part of the kernel API or a userspace convention?

I found the following but it is still not very clear to me:
* tcm_info field defined at [uapi/linux/rtnetlink.h#L616](https://github.com/torvalds/linux/blob/master/include/uapi/linux/rtnetlink.h#L616)
* macro TC_H_MAKE (used to set the tcm_info field) defined at [uapi/linux/pkt_sched.h#L72](https://github.com/torvalds/linux/blob/v6.2/include/uapi/linux/pkt_sched.h#L72)

Reply:
I don't know how it works from the kernel point of view. I used the same value as in https://github.com/florianl/tc-skeleton/blob/a40a7f0cad0d5eeded6e17377618c18458d84394/main.go#L136. Also, https://github.com/pantoniou/iproute2/blob/f443565f8df65e7d3b3e7cb5f4e94aec1e12d067/tc/tc_filter.c#L147 indicates it's `priority << 16 | proto`. Proto is `htons(ETH_P_ALL)`.